lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3649,7 +3650,11 @@ namespace {
         return nullptr;
 
       DeclName name = accessorInfo ? DeclName() : importedName.getDeclName();
-      if (importedName.importAsMember()) {
+
+      auto *enumDecl = dyn_cast<EnumDecl>(dc);
+      if (importedName.importAsMember() ||
+          (enumDecl &&
+           dyn_cast_or_null<clang::NamespaceDecl>(enumDecl->getClangDecl()))) {
```

Comment:
This still seems overly hardcoded. Maybe just `!dc->isModuleScopeContext()`? That should cover both cases, really. (If you're feeling iffy about it, you could assert the expected cases below.)

Reply:
Still requires a carve out for CXXMethodDecl. I tried to make CXXMethodDecl go through the import global as member pathway, but I found that it was not compatible. Namely, that code expects that the self parameter shows up as a member of the parameter list of the FunctionDecl. For CXXMethodDecl, this is entirely implicit.

Reply:
I guess that makes sense. We could choose whether the CXXMethodDecl handling should move to `importGlobalAsMethod` (although the name isn't really great) or leave it where it is now, but either way I like this better than special-casing NamespaceDecls. Thanks!

Reply:
If importGlobalAsMethod is changed to be able to import CXXMethodDecl, it will be capable of doing everything that this current function can do. In this case, it will not be necessary to have two functions. I can sketch that out in a followup if you'd like?

Reply:
Interesting. I suppose that's *already* true in the absence of CXXMethodDecl. `importGlobalAsInitializer` is still special, but everything else could be done here. Sounds good!